strapi/strapi
# [TS] Adds types for the params' filters object

### What does it do?

- Add type utility for the { filters } (params)
	- Only has an object notation
	- Can be used with or without generated types

### Why is it needed?

- Part of the filters typing initiative

### How to test it?

- Head to `packages/core/strapi/lib/services/entity-service/types/params/index.d.ts` and rename the file to `index.ts`
- Adds the following code snippet
  ```ts
	const filters: Filters.Any<Common.UID.Schema> = {
	  // ...
	}
	```
- And play around with the object, the types should be loose and accept either operators or nested fake attributes
- Then, generate types for the getstarted example project
- Add those lines to the top of the file you're in
	```ts
	import '../../../../../../../../examples/getstarted/types/generated/contentTypes'
	import '../../../../../../../../examples/getstarted/types/generated/components'
	```
- Modify the declaration of your filters by replacing `Common.UID.Schema` by an actual UID from a schema that has been generated
- Play around with the object, this time you should have strongly typed properties

### What's left?

I'm still unsure about how to deal with the ID since it's not a real attribute. If you have any idea, please share :slightly_smiling_face:
